moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -590,141 +590,194 @@ bool MoveItConfigData::outputROSControllersYAML(const std::string& file_path)
+            else
+              emitter << (*joint_it)->getName();
+          }
+          emitter << YAML::EndSeq;
+        }
+        else
+        {
+          emitter << YAML::Value << YAML::BeginMap;
+          emitter << joint_models[0]->getName();
+          emitter << YAML::EndMap;
+        }
+      }
+      emitter << YAML::Key << "sim_control_mode";
+      emitter << YAML::Value << "1";
+      emitter << YAML::Comment("0: position, 1: velocity");
+      emitter << YAML::Newline;
+      emitter << YAML::EndMap;
+    }
+    // Joint State Controller
+    emitter << YAML::Comment("Publish all joint states");
+    emitter << YAML::Newline << YAML::Comment("Creates the /joint_states topic necessary in ROS");
+    emitter << YAML::Key << "joint_state_controller" << YAML::Value << YAML::BeginMap;
     {
-      emitter << YAML::BeginMap;
-      emitter << YAML::Key << planning_groups[i][j] << YAML::Value << YAML::BeginMap;
-      emitter << YAML::Key << "trajectory";
-      emitter << YAML::Value << "0.60";
-      emitter << YAML::Key << "goal";
-      emitter << YAML::Value << "0.15";
+      emitter << YAML::Key << "type";
+      emitter << YAML::Value << "joint_state_controller/JointStateController";
+      emitter << YAML::Key << "publish_rate";
+      emitter << YAML::Value << "50";
       emitter << YAML::EndMap;
+    }
+    // Write default controllers
+    emitter << YAML::Key << "controller_list" << YAML::Value << YAML::BeginMap;
+    {
+      for (std::vector<Controller>::iterator controller_it = controllers_config_.begin();
+           controller_it != controllers_config_.end();)
+      {
+        // Depending on the controller type, fill the required data
+        if (controller_it->type_ == "FollowJointTrajectory")
+        {
+          emitter << YAML::Key << "name";
+          emitter << YAML::Value << controller_it->name_ + "_controller";
+          emitter << YAML::Key << "action_ns";
+          emitter << YAML::Value << "follow_joint_trajectory";
+          emitter << YAML::Key << "default";
+          emitter << YAML::Value << "True";
+          emitter << YAML::Key << "type";
+          emitter << YAML::Value << controller_it->type_;
+          // Write joints
+          emitter << YAML::Key << "joints";
+          {
+            joints_count_ = controller_it->joints_.size();
+            if (joints_count_ != 1)
+            {
+              emitter << YAML::Value << YAML::BeginSeq;
+
+              // Iterate through the joints
+              for (std::vector<std::string>::iterator joint_it = controller_it->joints_.begin();
+                   joint_it != controller_it->joints_.end(); ++joint_it)
+              {
+                emitter << *joint_it;
+              }
+              emitter << YAML::EndSeq;
+            }
+            else
+            {
+              emitter << YAML::Value << YAML::BeginMap;
+              emitter << controller_it->joints_[0];
+              emitter << YAML::EndMap;
```

Comment:
No I can't, because I found that if the `joints` were more than just `1` joint, they have to be written in the YAML as a `seq` and a `map` if only `1`. other than that `ros_control` doesn't read it right.